src/olympia/lib/git.py

```diff
@@ -346,6 +366,9 @@ def get_root_tree(self, commit):
 
         This doesn't contain the ``EXTRACTED_PREFIX`` prefix folder.
         """
+        if isinstance(commit, six.string_types):
+            commit = self.git_repository[commit]
```

Comment:
When does this happen?

Reply:
Added a comment in 58e3ce9ba2, it basically happens in the drf api serializers which simply grab `version.git_hash` and pass that on as `commit`

When that function is used internally (to hide the `extracted/` folder implementation detail) we are passing along actual commit objects.